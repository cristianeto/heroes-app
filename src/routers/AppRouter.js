import React, { useContext } from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';

import { AuthContext } from '../auth/AuthContext';
import PrivateRoute from './PrivateRoute';
import PublicRoute from './PublicRoute';

import DashboardRoutes from './DashboardRoutes';
import LoginScreen from '../components/login/LoginScreen';

const AppRouter = () => {
  const { user } = useContext(AuthContext);
  // console.log(user);
  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            exact
            path='/login'
            component={LoginScreen}
            isAuthenticated={user.logged}
          />

          <PrivateRoute
            path='/'
            component={DashboardRoutes}
            isAuthenticated={user.logged}
          />
        </Switch>
      </div>
    </Router>
  );
};

export default AppRouter;
