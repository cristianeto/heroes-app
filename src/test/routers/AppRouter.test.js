import { mount } from 'enzyme';
import { AuthContext } from '../../auth/AuthContext';
import AppRouter from '../../routers/AppRouter';

describe('Testing in <AppRouter/>', () => {
  const contextValue = { user: { logged: false }, dispatch: jest.fn() };

  test('should show my login component if user is not authenticated', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    // console.log(wrapper.find('h1').exists());
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').exists()).toBe(true);
  });

  test('should show a marvel component if user is authenticated', () => {
    const contextValue = {
      user: { name: 'cris', logged: true },
      dispatch: jest.fn(),
    };
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );
    // console.log(wrapper.html());
    expect(wrapper.find('.navbar').exists()).toBe(true);
  });
});
