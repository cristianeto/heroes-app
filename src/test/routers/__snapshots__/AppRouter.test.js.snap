// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <AppRouter/> should show my login component if user is not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      LoginScreen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
